ui/pages/token-allowance/token-allowance.js

```diff
@@ -208,6 +212,38 @@ export default function TokenAllowance({
 
   const isEmpty = customTokenAmount === '';
 
+  const network = hexToDecimal(fullTxData.chainId);
+
+  const currentNetworkUnapprovedTxs = Object.keys(unapprovedTxs)
+    .filter((key) =>
+      transactionMatchesNetwork(
+        unapprovedTxs[key],
+        fullTxData.chainId,
+        network,
+      ),
+    )
+    .reduce((acc, key) => ({ ...acc, [key]: unapprovedTxs[key] }), {});
+
+  const enumUnapprovedTxs = Object.keys(currentNetworkUnapprovedTxs);
+  const currentPosition = enumUnapprovedTxs.indexOf(
+    fullTxData.id ? fullTxData.id.toString() : '',
+  );
+
+  const totalTx = enumUnapprovedTxs.length;
+  const positionOfCurrentTx = currentPosition + 1;
+  const nextTxId = enumUnapprovedTxs[currentPosition + 1];
+  const prevTxId = enumUnapprovedTxs[currentPosition - 1];
+  const showNavigation = enumUnapprovedTxs.length > 1;
+  const firstTx = enumUnapprovedTxs[0];
+  const lastTx = enumUnapprovedTxs[enumUnapprovedTxs.length - 1];
+
+  const handleNextTx = (txId) => {
+    if (txId) {
+      dispatch(clearConfirmTransaction());
+      history.push(`${CONFIRM_TRANSACTION_ROUTE}/${txId}`);
+    }
+  };
```

Comment:
I don't think this is problematic but what I'm wondering is if there's an opportunity to improve the readability of our code by relocating and consolidating this logic and the 'confirm-transaction-base.container.js' files logic and move it *into* the `<ConfirmPageContainerNavigation>` component which appears to already be a functional component. Doing this would make that specific part of our code much more understandable and perhaps prevent needing to do this selector logic in multiple places.

Could you take a look into this and see how much effort it'll be to consolidate? Thanks @filipsekulic

Reply:
Hey @brad-decker! Good suggestion!
I refactored the code and you can see the changes here: f4b09382face3e44f085f59f4551ef4c5fb99272
However, there is a problem and you can see it in the attached screenshot, which is the reason why the pipeline is failing (`e2e test`). 
This happens when there are multiple token approvals and when a user confirms one of them, the navigation disappears for some reason and throws the error (screenshot attached). I think the problem is in the `confirm-page-container.component.js` within the `async componentDidMount`, which the error message is also suggesting (`...cancel all subscriptions and asynchronous tasks...`). 
I would appreciate if you could take a look and give your opinion on this one.

<img width="825" alt="Screenshot 2022-12-15 at 14 01 42" src="https://user-images.githubusercontent.com/92310504/207868219-d06ffbf8-c6da-4c3b-aba6-b81ddf981edd.png">

Reply:
Everything is fixed now - https://github.com/MetaMask/metamask-extension/pull/16822#discussion_r1053301032.